aws-rds-cfn-test-common/src/main/java/software/amazon/rds/test/common/schema/ResourceDriftTestHelper.java

```diff
@@ -0,0 +1,139 @@
+
+import com.api.jsonata4java.expressions.EvaluateException;
+import com.api.jsonata4java.expressions.Expressions;
+import com.api.jsonata4java.expressions.ParseException;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableSet;
+
+public class ResourceDriftTestHelper {
+
+    private final static Set<Class<?>> primitiveTypes = ImmutableSet.of(
+            Integer.class,
+            String.class,
+            Boolean.class
+    );
+
+    private static final String PROPERTIES = "/properties";
+    private static final String PROPERTY_OR_SPLIT_REGEX = "\\s*\\$OR\\s*";
+    private static final String PROPERTY_PATH_SEPARATOR = "/";
+
+    public static void assertResourceNotDrifted(
+            final Object input,
+            final Object output,
+            final JSONObject resourceSchema
+    ) {
+        assertResourceNotDrifted(input, output, resourceSchema, PROPERTIES);
+    }
+
+    public static void assertResourceNotDrifted(
+            final Object input,
+            final Object output,
+            final JSONObject resourceSchema,
+            final String basePath
+    ) {
+        if (input == null || output == null) {
+            Assertions.assertThat(input).isEqualTo(output);
+            return;
+        }
+        Assertions.assertThat(input).hasSameClassAs(output);
+
+        final JSONObject propertyTransformMap = resourceSchema.getJSONObject("propertyTransform");
+        final ObjectMapper objectMapper = new ObjectMapper();
+
+        JsonNode inputJsonNode = null;
+        try {
+            // JSONata accepts JsonNode as an input. The easiest way to get a JsonNode from an object
+            // is to convert it to a string and parse immediately.
+            inputJsonNode = objectMapper.readTree(objectMapper.writeValueAsString(input));
+        } catch (JsonProcessingException e) {
+            Assertions.fail(e.getMessage());
+        }
+
+        final Field[] fields = input.getClass().getDeclaredFields();
+
+        NextField:
+        for (final Field field : fields) {
+            // This code assumes that all data fields will have a @JsonProperty annotation.
+            final JsonProperty annotation = field.getAnnotation(JsonProperty.class);
+            if (annotation == null) {
+                continue;
+            }
+            final String fieldName = annotation.value();
+            field.setAccessible(true);
+            Object inputFieldVal = null;
+            Object outputFieldVal = null;
+            try {
+                inputFieldVal = field.get(input);
+                outputFieldVal = field.get(output);
+            } catch (IllegalAccessException e) {
+                Assertions.fail(e.getMessage());
+            }
+            if (Objects.equals(inputFieldVal, outputFieldVal)) {
```

Comment:
If the field is missing in `input`, but is present in `output` that is also not considered as drift. Exception to it is arrays.

Reply:
In this case i would prefer to keep the original logic in place. The purpose of this tool is not to completely emulate CFN behavior with all its conventions, but rather test the logic contained in `propertyTransform`. In this case we are risking of creating a false-positive check scenario when we've gotten a malformed input. Given these tests won't provide any specific value (i.e. it won't prove or disprove the logic), we could ignore this "relaxed" behavior of the drift detector.